src/main/java/htsjdk/samtools/util/CigarUtil.java

```diff
@@ -201,6 +212,13 @@ public static void clip3PrimeEndOfRead(SAMRecord rec, final int clipFrom, final
             }
         }
 
+        if (rec.getReadLength() != originalReadLength) {
```

Comment:
Could you add a method comment saying that this will remove these tags?

Reply:
done